stl/inc/memory

```diff
@@ -1506,6 +1542,10 @@ public:
 
     ~_Ref_count_obj2() {

```

Comment:
=default?

Reply:
No, that's what the newly added comment citing N4849 [class.dtor]/7 is explaining. Attempting to `= default` this will delete it instead, because of the `union`.

Reply:
Please meow if you want me to change the comments to be even more explicit about "this is why we can't =default".

Reply:
I assume there's a test that would fail if this were changed to =default which is superior to a comment anyway.

Reply:
Yep - massive, unmissable compiler errors. I encountered them and added the comments.